### PR TITLE
feat(datepicker): add suppression of validation for #3836

### PR DESCRIPTION
**Pull request by Blackbaud-PatrickOFriel**
This adds the `$datepickerSuppressError` value provider which has an `isSuppressed` property to allow the option to suppress the `$log` error in `DatepickerController`. This was made to address #3836

### CODE REVIEW

File: src/datepicker/datepicker.js [hunk 1]
```diff
@@ -1,5 +1,9 @@
 angular.module('ui.bootstrap.datepicker', ['ui.bootstrap.dateparser', 'ui.bootstrap.position'])
 
+.value('$datepickerSuppressError', {
```
Comment (wesleycho) [inline src/datepicker/datepicker.js]: Just make it a boolean - no need for an object here.

**Comment by wesleycho** [comment]
Just make all changes related to my comment and this PR should be good to get in.

**Comment by Blackbaud-PatrickOFriel** [comment]
The only reason I made it an object is because when it's a literal, changes you make upon injection don't take effect (setting `$datepickerSuppressError` to `true`). You could only change the value in the config phase using `$provide`. If that's the route I should take, then I'll have to make the test a bit different since you can't use `$provide` after you've used `inject`.

**Comment by wesleycho** [comment]
I think that is fine - there really isn't a great use case for toggling suppression on or off once the app is running.

**Comment by Blackbaud-PatrickOFriel** [comment]
Okay I'll go ahead and make the change then

**Comment by Blackbaud-PatrickOFriel** [comment]
So in order to test properly I had to wrap the other tests in their own describe block so that I could override the `$provide` for my test cases before the `inject` function was ever called. And now the diff tool makes it look like I deleted everything and added it back...

**Comment by wesleycho** [comment]
That's perfectly fine

**Comment by wesleycho** [comment]
Can you squash your commits into one?

**Comment by Blackbaud-PatrickOFriel** [comment]
Squash complete

**Comment by wesleycho** [comment]
LGTM - merging shortly. Thanks for the work!

**Comment by Blackbaud-PatrickOFriel** [comment]
No problem! Thanks for this entire library, it's great!

**Pull request closed without merging**